File: backend/schemas/LocationPermission.py
"""
Pydantic schema for LocationPermission for response and request validation
"""

from pydantic import BaseModel
from datetime import datetime


class CreateLocationPermission(BaseModel):
    """Schema for creating a location permission"""
    user_id: int


class LocationPermissionResponse(BaseModel):
    """Schema for location permission response"""
    loc_perm_id: int
    user_loc_id: int
    user_id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/lib/models/location_permission.dart
class LocationPermission {
  final int locPermId;
  final int userLocId;
  final int userId;
  final DateTime createdAt;
  final DateTime updatedAt;

  LocationPermission({
    required this.locPermId,
    required this.userLocId,
    required this.userId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory LocationPermission.fromJson(Map<String, dynamic> json) {
    return LocationPermission(
      locPermId: json['loc_perm_id'] as int,
      userLocId: json['user_loc_id'] as int,
      userId: json['user_id'] as int,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'loc_perm_id': locPermId,
      'user_loc_id': userLocId,
      'user_id': userId,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }
}
